tests/test.py

```diff
@@ -0,0 +1,101 @@
+import unittest
+from homework import Rectangle
+import math
+
+
+class TestInvalidRectangleProperties(unittest.TestCase):
+
+    def setUp(self):
+        self.rect = Rectangle
+        self.invalid_args = ((0, 0), (-1, -1), (-1, 0),
+                             (0, -1), (1, 0), (0, 1),
+                             (1, -1), (-1, 1))
+
+    def tearDown(self) -> None:
+        del self.rect
+
+    @unittest.skip("This version of Rectangle class doesn't support assessment of values")
+    def test_invalid_params(self):
+        for args in self.invalid_args:
+            with self.subTest(args=args):
+                self.assertRaises(ValueError, self.rect, *args)
+
+
+class TestValidRectangleProperties(unittest.TestCase):
+
+    def setUp(self):
+        self.rect = Rectangle
+
+    def tearDown(self) -> None:
+        del self.rect
+
+    def test_unexpected_errors(self):
+        for args in zip(range(1, 8, 2), range(8, 1, -2)):
+            try:
+                self.rect(*args)
+            except Exception as er:
+                self.fail(f'{type(er)} happened:  with args: {args}')
+
+
+class TestRectangleMethods(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.rect = Rectangle(2, 4)
+        self.sqr = Rectangle(2, 2)
+        self.ctrl_rect_diag = math.sqrt(
+                                        math.pow(self.rect.height, 2) +
+                                        math.pow(self.rect.width, 2))
+        self.ctrl_sqr_diag = math.sqrt(
+                                        math.pow(self.sqr.height, 2) +
+                                        math.pow(self.sqr.width, 2))
+
+    def tearDown(self) -> None:
+        del self.rect
+        del self.sqr
+
+    def test_get_rectangle_perimeter(self):
+        self.assertEqual(self.rect.get_rectangle_perimeter(), 12)
+
+    def test_get_rectangle_square(self):
+        self.assertEqual(self.rect.get_rectangle_square(), 8)
+
+    def test_get_sum_of_corners(self):
+        sum_of_corn = self.rect.get_sum_of_corners
+        invalid_num_of_corn = [-1, 0, 5]
+
+        for val in invalid_num_of_corn:
+
+            with self.subTest(val=val):
+                if val in [-1, 0]:
+                    self.skipTest("The method doesn't check either zero or negative values")
```

Comment:
if method doesn't support some values, it's reason to make function refactoring. no need to create test that will be skipped by default.

Reply:
Done